src/overlays/actors/ovl_En_Xc/z_en_xc.c

```diff
@@ -77,479 +47,2365 @@ static ColliderCylinderInitType1 sCylinderInit = {
+                    EnXc_SpawnFlame(this, globalCtx);
+                }
+
+                if (action == 1) {
+                    EnXc_DestroyFlame(this);
+                }
+
+                D_80B41DA8 = action;
+            }
+
+            EnXc_SetupFlamePos(this, globalCtx);
+        }
+    }
+}
+
+void func_80B3D48C(EnXc* this, GlobalContext* globalCtx) {
+    CutsceneContext* csCtx = &globalCtx->csCtx;
+    CsCmdActorAction* linkAction = csCtx->linkAction;
+    s16 yaw;
+
+    if (linkAction != NULL) {
+        yaw = (u16)linkAction->rot.y + 0x8000;
+    } else {
+        Player* player = PLAYER;
+        yaw = player->actor.world.rot.y + 0x8000;
+    }
+
+    this->actor.world.rot.y = yaw;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+}
+
+AnimationHeader* EnXc_GetCurrentHarpAnim(GlobalContext* globalCtx, s32 index) {
+    AnimationHeader* animation = &gShiekPlayingHarp5Anim;
+    CsCmdActorAction* npcAction = EnXc_GetCsCmd(globalCtx, index);
+
+    if (npcAction != NULL) {
+        u16 action = npcAction->action;
+        if (action == 11) {
+            animation = &gShiekPlayingHarp3Anim;
+        } else if (action == 12) {
+            animation = &gShiekPlayingHarp2Anim;
+        } else if (action == 13) {
+            animation = &gShiekPlayingHarp4Anim;
+        } else {
+            animation = &gShiekPlayingHarp5Anim;
+            if (action == 23) {
+                animation = &gShiekPlayingHarpAnim;
+            }
+        }
+    }
+    return animation;
+}
+
+void EnXc_CalcXZAccel(EnXc* this) {
+    f32 timer = this->timer;
+    f32* speedXZ = &this->actor.speedXZ;
+
+    if (timer < 9.0f) {
+        *speedXZ = 0.0f;
+    } else {
+        if (timer < 3.0f) {
+            *speedXZ = (((kREG(2) * 0.01f) + 1.2f) / 3.0f) * (timer - 9.0f);
+        } else {
+            *speedXZ = (kREG(2) * 0.01f) + 1.2f;
+        }
+    }
+    Actor_MoveForward(&this->actor);
+}
+
+void func_80B3D644(EnXc* this) {
+    Actor_MoveForward(&this->actor);
+}
+
+void EnXc_CalcXZSpeed(EnXc* this) {
+    f32 timer = this->timer;
+    f32* speedXZ = &this->actor.speedXZ;
+
+    if (timer < 3.0f) {
+        *speedXZ = (((kREG(2) * 0.01f) + 1.2f) / 3.0f) * (3.0f - timer);
+    } else {
+        *speedXZ = 0.0f;
+    }
+    Actor_MoveForward(&this->actor);
+}
+
+void func_80B3D6F0(EnXc* this) {
+    EnXc_CalcXZAccel(this);
+}
+
+void func_80B3D710(EnXc* this) {
+    Actor_MoveForward(&this->actor);
+}
+
+void func_80B3D730(EnXc* this) {
+    EnXc_CalcXZSpeed(this);
+}
+
+void func_80B3D750(EnXc* this, GlobalContext* globalCtx) {
+    if (EnXc_MinuetCS(this, globalCtx) && EnXc_BoleroCS(this, globalCtx)) {
+        this->action = SHIEK_ACTION_WAIT;
+    }
+}
+
+void EnXc_SetupFallFromSkyAction(EnXc* this, GlobalContext* globalCtx) {
+    s32 pad;
+    CutsceneContext* csCtx = &globalCtx->csCtx;
+    CsCmdActorAction* npcAction;
+
+    if (csCtx->state && (npcAction = csCtx->npcActions[4], npcAction && npcAction->action == 2)) {
```

Comment:
This should probably be split in 2 ifs so we avoid using the comma operator.
Relevant in other places in this file